src/context/product-context.js

```diff
@@ -1,10 +1,17 @@
-import { createContext, useContext, useEffect, useState } from "react";
+import { createContext, useContext, useEffect, useReducer } from "react";
 import axios from "axios";
+import productReducer from "../reducer/product-reducer";
 
 const ProductContext = createContext();
 
+const products = [];
+const noProducts = 0;
+
 export function ProductProvider({ children }) {
-  const [products, setProducts] = useState([]);
+  const [productState, productDispatch] = useReducer(productReducer, {
+    products,
+    noProducts,
+  });
 
   async function getProducts() {
```

Comment:
You can use arrow functions,just a suggestion, although it doesn't make a functional difference.

Reply:
yess, I will do this in a refactoring PR